doc/fluidsettings.xml

```diff
@@ -590,6 +590,15 @@ and commit the results. Refresh with the following command:
                 Sets the performance mode as pointed out by Oboe's documentation.
             </desc>
         </setting>
+        <setting>
+            <name>oboe.error-recovery-mode</name>
```

Comment:
There is a mismatch: The setting is called `audio.oboe.audio-error-recovery-mode` in the code.

On the other hand, I don't think a driver specific setting will be necessary. Maybe we should promote this to a general setting right away. Something like `audio.error-recovery-mode` sounds good to me. This way it could be used by any driver.

In case you insist on a driver specific setting, I would vote for calling it `audio.oboe.error-recovery-mode`.

Reply:
Thanks for the review. I fixed all other issues, and I am all for making error recovery generic.

Though I think it should rather be done in common API level, not just in Oboe driver (implementation on those unsupported platforms can be left blank). Automatic reconnection is kind of hack and app developers, not driver developers (us), would rather like to get notified so that they can control the situation in their own way, not just by choosing a configuration option.

Would it sound good if we add a new error callback setter on `fluid_audio_driver_t` ? like:

```
typedef void (*fluid_audio_driver_error_callback_t)(fluid_audio_driver_t*, void*);
fluid_audio_driver_set_error_callback(fluid_audio_driver_t* driver, fluid_audio_driver_error_callback_t func, void* data);
```

Current settings-based error handler may be still useful, but at the sane time I'm not very inclined to move "Reconnect" behavior onto driver-agnostic plate, as (1) it is a hacky behavior, and (2) it requires driver-specific implementation. I'm just not opinionated and it can be implemented, left as is though.

Reply:
Some investigation I made on device change callbacks:

- Apple https://developer.apple.com/library/archive/documentation/Audio/Conceptual/AudioSessionProgrammingGuide/HandlingAudioHardwareRouteChanges/HandlingAudioHardwareRouteChanges.html
- cannot find anything on win32. On wasapi I found this relevant https://social.msdn.microsoft.com/Forums/en-US/c145b1ce-187f-4b08-9a8b-b4baacf6591d/change-playback-device-of-an-existing-wasapi-audiosession
- UWP https://docs.microsoft.com/en-us/uwp/api/windows.media.devices.mediadevice.defaultaudiorenderdevicechanged?view=winrt-20348
- not sure about alsa/jack/pulseaudio etc. either.
- JUCE has error callback support in `juce_audio_devices` API https://docs.juce.com/master/classAudioIODeviceCallback.html#a7768d640daf7d5d56399ced8188b2b75
- there is an ongoing discussion at miniaudio https://github.com/mackron/miniaudio/issues/325

Reply:
I'm struggling with the callback function. Yes, it seems like a good idea and goes hand in hand with an idea I had in #928, but it also raises many questions:
* In which cases do you get an error callback?
  * Device disconnected?
  * Audio Server executable crashed? (think of pulseaudio or jack)
  * Buffer underrun??
* In which state will the driver be after an error occurred?
  *  Will it be a recoverable state?
  * If yes, how should the client app recover from it? The client app doesn't know anything about the internals of the driver.

This all is actually quite beyond the scope of this PR. So, I'd say to keep it as driver specific setting
 `audio.oboe.error-recovery-mode` for now. Because changing a driver specific setting in the future should be less painful than a general setting.

Reply:
Agreed, let's make this PR simpler. It is now renamed in code.